### PR TITLE
Minapp 1 add basic classes

### PR DESCRIPTION
**Pull request by MartinBartosek**
Added controller class, service classes and did some refactoring.

### CODE REVIEW

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    public Integer calculate(String expression) throws MultipleOperatorException {
+
+        char[] characters = expression.toCharArray();
+        Stack<Integer> numbers = new Stack<Integer>();
+        Stack<Character> operators = new Stack<Character>();
+
+        for (int i = 0; i < characters.length; i++) {
+            if (characters[i] == ' ') {
+                continue;
+            }
```
Comment (nlobanov) [inline src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java]: Any reason for this?

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorService.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.pillarglobal.miniApp.Calculator.controllers.services;
+
+import com.pillarglobal.miniApp.Calculator.exceptions.MultipleOperatorException;
+
+public interface CalculatorService {
```
Comment (nlobanov) [inline src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorService.java:5]: Do we need this interface?

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+import com.pillarglobal.miniApp.Calculator.exceptions.MultipleOperatorException;
+import org.springframework.stereotype.Service;
+
+import java.util.Stack;
+
+@Service
+public class CalculatorServiceImpl implements CalculatorService {
+
+    public Integer calculate(String expression) throws MultipleOperatorException {
```
Comment (nlobanov) [inline src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java]: Calculate does parsing? Why do we call it calculate then?

File: src/test/java/com/pillarglobal/miniApp/services/CalculatorServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    void calculateExpression() {
+    }
+
+    @Test
+    void hasPrecedence() {
+    }
+
+    @Test
+    void applyOperations() {
+    }
```
Comment (nlobanov) [inline src/test/java/com/pillarglobal/miniApp/services/CalculatorServiceImplTest.java:22]: Add tests to be sure parsing works

**Review by nlobanov — CHANGES_REQUESTED** [review]

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java (hunk 1 above)
Comment (MartinBartosek) [reply src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java]: Removed, I thought I need to check spaces in expression but it seems to ignore them.

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorService.java (hunk 2 above)
Comment (MartinBartosek) [reply src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorService.java:5]: well, I will leave it there for future stuff :D Right now, it's not necessary, true, but as I said.

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java (hunk 3 above)
Comment (MartinBartosek) [reply src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java]: I renamed it to solveExpression. Hope it's OK.

File: src/test/java/com/pillarglobal/miniApp/services/CalculatorServiceImplTest.java (hunk 4 above)
Comment (MartinBartosek) [reply src/test/java/com/pillarglobal/miniApp/services/CalculatorServiceImplTest.java:22]: Test are on the way right know in the next branch/pull request.

File: src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java (hunk 3 above)
Comment (nlobanov) [reply src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java]: Should we rename other things that have the word calculate in their name as well?

Comment (MartinBartosek) [reply src/main/java/com/pillarglobal/miniApp/Calculator/controllers/services/CalculatorServiceImpl.java]: Done.

**Review by nlobanov — APPROVED** [review]

**Pull request merged**